clients/wallet/src/task.rs

```diff
@@ -0,0 +1,206 @@
+use crate::Ledger;
+use tokio::sync::oneshot::{channel, error::TryRecvError, Receiver, Sender};
+
+pub const LEDGER_TASK_MAX_RETRIES: u8 = 3;
+/// Determines the status of the task of processing a transaction
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum LedgerTaskStatus {
+	NotStarted,
+	/// Process Done and it was a success
+	ProcessSuccess,
+	/// The error is acceptable, and is best to just retry again
+	RecoverableError,
+	/// Something happened when
+	Error(String),
+}
+
+pub struct LedgerTask {
+	ledger: Ledger,
+
+	/// receives the status of the task execution
+	task_execution_receiver: Receiver<LedgerTaskStatus>,
+
+	/// the number of retries allowed to process this specific task
+	retries_remaining: u8,
+
+	/// The latest status of this task
+	latest_status: LedgerTaskStatus,
+}
+
+impl LedgerTask {
+	/// Creates a task structure the `TransactionEnvelope`
```

Comment:
Not sure if a word is missing here

	/// Creates a task structure **for** the `TransactionEnvelope`

Reply:
updated, 🙇